src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.cs

```diff
@@ -250,36 +252,30 @@ private void CommitItem(CompletionItem item)
         }
 
         private const int MaxMRUSize = 10;
-        private ImmutableArray<string> _recentItems = ImmutableArray<string>.Empty;
+        private ImmutableList<string> _recentItems = ImmutableList<string>.Empty;
 
         public string DisplayName => EditorFeaturesResources.Code_Completion;
 
         public void MakeMostRecentItem(string item)
         {
-            bool updated = false;
-
-            while (!updated)
-            {
-                var oldItems = _recentItems;
-
+            ImmutableInterlocked.Update(ref _recentItems, (ImmutableList<string> oldItems) => {
```

Comment:
nit: parameter type for the lambda seems superfluous.  Remove unless you feel it really adds to the clarity (i personally don't tink it does).

Reply:
also, this is the most unnecessary usage of InterlockedUpdate i've ever seen.  We can totally just grab the value, slam in the new value, and stash the old array away... but htat's not really relevant to this review.